public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx

```diff
@@ -0,0 +1,407 @@
+            unitNames = unitNames.filter(n => n.toLowerCase() !== 'no unit');
+            if (noUnit.length > 0) {
+                unitNames.push(noUnit[0]);
+            }
+            if (unitNames.length > 0) {
+                    this.setState({
+                        selectedUnit: unitNames[0],
+                        unitNames: unitNames,
+                        genderResults: genderResults
+                    }, () => {
+                        this.showMeasurementGenderHistogram(unitNames[0]);
+                    });
+            }
+        }
+        return;
+    }
+
+    showMeasurementGenderHistogram(unit: string) {
+        const {selectedChartAnalysis, measurementGenderCountAnalysis} = this.state;
+        let mixtureOfValues = false;
+        if (unit.toLowerCase() === 'no unit') {
+            const unitResults = selectedChartAnalysis.find(aa => aa.unitName === unit);
+            if (unitResults && unitResults.results && unitResults.results.length > 0) {
+                const numericResults = unitResults.results.filter(r => r.measurementValueType === 'numeric');
+                const textResults = unitResults.results.filter(r => r.measurementValueType === 'text');
+                if (numericResults && numericResults.length > 0 && textResults && textResults.length > 0) {
+                    mixtureOfValues = true;
+                }
+            }
+        }
+        const toDisplayMeasurementGenderAnalysis = { ...selectedChartAnalysis.find(aa => aa.unitName === unit) };
+        let toDisplayMeasurementGenderCountAnalysis = null;
+        if (measurementGenderCountAnalysis) {
+            toDisplayMeasurementGenderCountAnalysis = measurementGenderCountAnalysis.find(aa => aa.unitName === unit);
+        }
+        let selectedMeasurementType = null;
+        if (mixtureOfValues) {
+            toDisplayMeasurementGenderAnalysis.results = toDisplayMeasurementGenderAnalysis.results.filter(r => r.measurementValueType === 'text');
+            selectedMeasurementType = 'No Unit (Text)';
+        }
+        this.setState({
+            selectedUnit: unit,
+            mixtureOfValues: mixtureOfValues,
+            selectedMeasurementType: selectedMeasurementType,
+            toDisplayMeasurementGenderAnalysis: toDisplayMeasurementGenderAnalysis,
+            toDisplayMeasurementGenderCountAnalysis: toDisplayMeasurementGenderCountAnalysis
+        });
+    }
+
+    fetchChartTitle(gender: any) {
+        const {toDisplayMeasurementGenderCountAnalysis} = this.state;
+        if (toDisplayMeasurementGenderCountAnalysis) {
+            const genderResults = toDisplayMeasurementGenderCountAnalysis.results
+                .filter(r => r.stratum3 === gender.stratum2)[0];
+            if (genderResults && genderResults.countValue > 20) {
+                return gender.analysisStratumName + ' - ' + genderResults.countValue;
+            } else {
+                return gender.analysisStratumName + ' - &le; ' + 20;
+            }
```

Comment:
Nit: Can be simplified to a single statement:
```suggestion
            return genderResults && genderResults.countValue > 20 ?
              gender.analysisStratumName + ' - ' + genderResults.countValue :
              gender.analysisStratumName + ' - &le; ' + 20;
```

Reply:
Changed this. thanks